module_utils/network_lsr/argument_validator.py

```diff
@@ -2012,3 +2071,30 @@ def _ipv6_is_not_configured(connection):
                             "match.path is not supported by the running version of "
                             "NetworkManger.",
                         )
+        self.validate_route_tables(connection, idx)
+
+
+class RouteTableParsing:
+    @classmethod
+    def parse_route_tables_mapping(cls, file_content):
+        table_cache = {}
+        for table in file_content.split("\n"):
+            table_mapping = table.split()
+            # This is the original code about the route tables parsing in iproute2,
+            # https://git.kernel.org/pub/scm/network/iproute2/iproute2.git/tree/lib/rt_names.c?id=ade99e208c1843ed3b6eb9d138aa15a6a5eb5219#n38
+            if len(table_mapping) == 2 and table[0] != "#" and table[0] != "0":
```

Comment:
iproute also support a `rt_tables.d`  directory. That seems rather useful, and I would think that all users should drop their aliases in their own files...

Anyway, the code here should make an effort to be exactly like what iproute2 does in the parsing.